apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -199,30 +246,42 @@ export default class AnimationPickerBody extends React.Component {
       onAnimationSelectionComplete
     } = this.props;
 
+    let paddingOffset = this.state.headerOffset;
+    if (searchQuery !== '' || categoryQuery !== '') {
+      paddingOffset += this.state.categoryHeight + this.state.footerHeight;
+    }
```

Comment:
i don't think we should set these values in state -- we should reference the refs directly unless react gives some warnings/errors about referencing refs within the render method (i don't know of any). i think this for a couple of reasons:
- this isn't state that the component needs to know about, rather some UI side-effects that we need to track
- we're deciding when to recalculate this state, which may not match when the DOM renders/resizes/etc those elements and could be error-prone

we may still want to abstract this calculation into a helper function to keep the render function readable, especially if there is some conditional logic (e.g., only setting footerHeight is multiselect is enabled)

Reply:
I think the issue is less about us referencing refs in the render method and more about us trying to calculate a value (offset) that impacts how the render should look. If my understanding is correct, if we put the calculation in the render, before the item is rendered, then the element hasn't been rendered yet, so it doesn't have a height. Putting this in the componentnDidMount waits until the item has been rendered once so the element being ref-ed exists and has a height we can look at. Then setting that as part of the state forces a re-render. With these limitations in mind, is there a better way to do this?

Reply:
Happy to hop on a call to talk more about this.